src/goto-instrument/full_slicer.cpp

```diff
@@ -185,7 +185,7 @@ void full_slicert::add_jumps(
         if(cfg[entry->second].node_required)
         {
           const irep_idt id2=goto_programt::get_function_id(*d_it);
-          assert(id==id2);
+          INVARIANT(id==id2, "explanation needed");
```

Comment:
The error message should be: "goto/jump expected to be within a single function"